warning in transport_codec.cpp

### Priority

Low

### Area

- [ ] Datapack
- [x] Source
- [ ] Map
- [ ] Other

### What happened?

When compiling Canary on modern Linux distributions (such as Ubuntu 24.04 or Debian 12) using newer GCC versions (GCC 13/14), a deprecation warning is triggered in `transport_codec.cpp`. 

The compilation completes successfully, but it outputs a warning regarding arithmetic operations between different enumeration types.

### Error Log / Warning

root@debian12:~/canary# cmake --build --preset linux-release -j4
[7/33] Building CXX object src/CMakeFiles/canary.dir/Unity/unity_24_cxx.cxx.o
In file included from /root/canary/build/linux-release/src/CMakeFiles/canary.dir/Unity/unity_24_cxx.cxx:13:
/root/canary/src/server/network/protocol/transport_codec.cpp: In member function ‘bool TransportCodec::decryptXtea(Protocol&, NetworkMessage&) const’:
/root/canary/src/server/network/protocol/transport_codec.cpp:117:129: warning: arithmetic between different enumeration types ‘<unnamed enum>’ and ‘<unnamed enum>’ is deprecated [-Wdeprecated-enum-enum-conversion]
  117 |         uint16_t msgLength = msg.getLength() - (protocol.checksumMethod == CHECKSUM_METHOD_NONE ? HEADER_LENGTH : HEADER_LENGTH + CHECKSUM_LENGTH);
      |                                                                                                                   ~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~

### What OS are you seeing the problem on?

Linux

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

--- a/src/server/network/message/networkmessage.cpp
+++ b/src/server/network/message/networkmessage.cpp
@@ -276,7 +276,6 @@ const uint8_t* NetworkMessage::getBuffer() const {
 uint8_t* NetworkMessage::getBodyBuffer() {
 	info.position = 2;
 	// Return the pointer to the body of the buffer starting after the header
-	// Convert HEADER_LENGTH to uintptr_t to ensure safe pointer arithmetic with enum type
 	return buffer.data() + static_cast<std::uintptr_t>(HEADER_LENGTH);
 }
 

--- a/src/server/server_definitions.hpp
+++ b/src/server/server_definitions.hpp
@@ -14,8 +14,8 @@
 // Enums
 // Connection and networkmessage.
 enum { FORCE_CLOSE = true };
-enum { HEADER_LENGTH = 2 };
-enum { CHECKSUM_LENGTH = 4 };
+static constexpr int32_t HEADER_LENGTH = 2;
+static constexpr int32_t CHECKSUM_LENGTH = 4;
 enum { XTEA_MULTIPLE = 8 };
 enum { MAX_BODY_LENGTH = NETWORKMESSAGE_MAXSIZE - HEADER_LENGTH - CHECKSUM_LENGTH - XTEA_MULTIPLE };
 enum { MAX_PROTOCOL_BODY_LENGTH = MAX_BODY_LENGTH - 10 };

--- a/src/server/server_definitions.hpp
+++ b/src/server/server_definitions.hpp
@@ -11,22 +11,21 @@
 
 #include "utils/const.hpp"
 
-// Enums
 // Connection and networkmessage.
-enum { FORCE_CLOSE = true };
+static constexpr bool FORCE_CLOSE = true;
 static constexpr int32_t HEADER_LENGTH = 2;
 static constexpr int32_t CHECKSUM_LENGTH = 4;
-enum { XTEA_MULTIPLE = 8 };
-enum { MAX_BODY_LENGTH = NETWORKMESSAGE_MAXSIZE - HEADER_LENGTH - CHECKSUM_LENGTH - XTEA_MULTIPLE };
-enum { MAX_PROTOCOL_BODY_LENGTH = MAX_BODY_LENGTH - 10 };
+static constexpr int32_t XTEA_MULTIPLE = 8;
+static constexpr int32_t MAX_BODY_LENGTH = NETWORKMESSAGE_MAXSIZE - HEADER_LENGTH - CHECKSUM_LENGTH - XTEA_MULTIPLE;
+static constexpr int32_t MAX_PROTOCOL_BODY_LENGTH = MAX_BODY_LENGTH - 10;
 
+// Enums
 enum ConnectionState_t : uint8_t {
 	CONNECTION_STATE_OPEN,
 	CONNECTION_STATE_IDENTIFYING,
 	CONNECTION_STATE_READINGS,
 	CONNECTION_STATE_CLOSED
 };
-// Connection and networkmessage.
 
 // Protocol.
 enum RequestedInfo_t : uint16_t {

